src/app/app.component.ts

```diff
@@ -6,6 +6,42 @@ import { Component } from '@angular/core';
   styleUrls: ['./app.component.scss'],
 })
 class AppComponent {
+  searchKeyword : string = '';
+
+  showSearchPage : boolean = false;
+
+  isSettingsOpen : boolean = false;
+
+  dateSortOrder : 'ascending' | 'descending' | null = null;
```

Comment:
```suggestion
  dateSortOrder? : string;
```

since `ascending` is not an actual type :)

Reply:
also you could create `enum` to store `asc/desc` values if needed/want to